From 4883f31786f2318e3a15f8a4c060dc4e5269f442 Mon Sep 17 00:00:00 2001
From: Wang Panzhenzhuan <randy.wang@rock-chips.com>
Date: Tue, 20 Nov 2018 09:46:57 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3399-sapphire-excavator-edp:
 distinguish isp relate nodes for Android 9.0 & 8.1

rk3399-sapphire-excavator-edp*:
isp0/1 for CameraHal1 in Android8.1 or lower version
rkisp1_0/1 & mipi_dphy_rx0/tx1rx1 for CameraHal3 in Android9.0

Change-Id: I6bae9306d22edaebc6aaf6a3f49129f26977e623
Signed-off-by: Wang Panzhenzhuan <randy.wang@rock-chips.com>
---
 .../rk3399-sapphire-excavator-edp-avb.dts       | 17 +++++++++++++++++
 .../rockchip/rk3399-sapphire-excavator-edp.dts  |  8 ++++++++
 .../rockchip/rk3399-sapphire-excavator-edp.dtsi |  8 --------
 3 files changed, 25 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp-avb.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp-avb.dts
index 79bab0a3d1b0..79855b5f489f 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp-avb.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp-avb.dts
@@ -32,3 +32,20 @@
 		};
 	};
 };
+
+&mipi_dphy_rx0 {
+	status = "okay";
+};
+
+&mipi_dphy_tx1rx1 {
+	status = "okay";
+};
+
+&rkisp1_0 {
+	status = "okay";
+};
+
+&rkisp1_1 {
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dts
index bd6356423c3b..725e39f10556 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dts
@@ -68,3 +68,11 @@
 		};
 	};
 };
+
+&isp0 {
+	status = "okay";
+};
+
+&isp1 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dtsi
index 2af6f3d6ca39..9a68b0cad631 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dtsi
@@ -255,14 +255,6 @@
 	};
 };
 
-&isp0 {
-	status = "okay";
-};
-
-&isp1 {
-	status = "okay";
-};
-
 &isp0_mmu {
 	status = "okay";
 };
-- 
2.35.3

